### ISSUE TITLE
checksums: Retrieve Artifact endpoint returns 404

### ISSUE THREAD

**Issue by jasonwalsh** [jfrog/artifactory-user-plugins#253]
I have installed and configured the `checksums` user plugin as defined in the README. However, whenever I attempt to issue an HTTP request to the Retrieve Artifact endpoint with the `.sha512` extension, the server responds with a 404 Not Found response. I understand that the plugin calculates the checksums lazily, but it does not appear to be even attempting to generate a new checksum using the SHA512 algorithm.

<img width="822" alt="screen shot 2018-11-27 at 10 42 50 am" src="https://user-images.githubusercontent.com/2184329/49109029-ab9bef80-f257-11e8-89f9-d01170d7f137.png">

The propagation of the plugin appears to have run successfully, so I am unsure as to what is causing the issue.

Am I understanding how this plugin works correctly?

**Comment by elioengcomp** [comment]
Hi @jasonwalsh,

In your logs I don't see the message showing the plugin has been loaded. Have you called the[ Reload Plugins API ](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-ReloadPlugins)after installing it?

**Comment by jasonwalsh** [comment]
Hey @elioengcomp I just tried that and the plugin reloaded successfully, but the HTTP request to the Retrieve Artifact endpoint still returns a 404.

    $ curl -u <user:password> https://artifactory.corp.mongodb.com/artifactory/npm/ajv/-/ajv-4.11.8.tgz.sha512

```json
{
  "errors" : [ {
    "status" : 404,
    "message" : "Could not find resource"
  } ]
}
```

The resource `ajv-4.11.8.tgz` does exit.

**Comment by jasonwalsh** [comment]
I managed to get it working. I should have mentioned that the environment is an HA cluster. I was making an HTTP request to the Reload Plugins endpoint via the load balancer instead of the leader. Once I issued the request to the leader, the plugin reloaded successfully and propagated to the other followers in the cluster.

I am not able to issue an HTTP request to the load balancer FQDN and calculate the SHA512 checksum of uploaded artifacts.

Thanks for your help!
